### PR TITLE
Feature collimator2 update

### PR DESCRIPTION
**Pull request by chandabindu**
Updated geometry for collimator2 to avoid edge scattering.

### CODE REVIEW

File: geometry/upstream/upstreamDaughter_merged.gdml [hunk 1]
```diff
@@ -350,88 +346,195 @@
...
+        </subtraction>
+
+        <subtraction name ="Coll2_Cu_S8">
+        <first ref="Coll2_Cu_S7"/>
+        <second ref="Coll2_Cu_S2"/>
+        <position name="pos_Cu_6" x="0" y="0" z="25" lunit="mm"/>
+        <rotation name="rot_Cu_6" x="0" y="0" z="SEPTANT*(5+0.75)" unit="deg"/>
+        </subtraction>
+
+        <subtraction name ="Coll2_Cu_S9">
```
Comment (wdconinc) [inline geometry/upstream/upstreamDaughter_merged.gdml:532]: The repeated subtractions are very slow, especially for collimators where many tracks are passing through. Can you implement this with nested volumes? Can this repeated subtraction live inside a vacuum volume based on the arb8 Coll2_Cu_S? At least then geant4 only has to deal with the subtractions when it is actually inside the arb8.

Comment (wdconinc) [reply geometry/upstream/upstreamDaughter_merged.gdml:532]: Of course, same applies to the other repeated subtraction as well. This is only possible if the arb8 has nothing 'sticking' into it.

Comment (wdconinc) [reply geometry/upstream/upstreamDaughter_merged.gdml:532]: @chandabindu

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: geometry/upstream/upstreamDaughter_merged.gdml (hunk 1 above)
Comment (chandabindu) [reply geometry/upstream/upstreamDaughter_merged.gdml:532]: If it  was okay with the previous Collimator 2 and collimator 4 design (especially number subtraction they had in their geometries), then the present implementation of collimator 2 should also be fine - there are only one extra-subtraction step involved for the new  collimator 2 stage (though the W-part of the collimator 2  now has two solids).

I ran simulations with 10K events (with same seed) with beam generators with 'develop' and with 'feature-collimator2-update' branch on ifarm1901..
Here is the run time statistics--

**For 'develop' branch**
G4WT0 > --> Event 9000 starts with initial seeds (45799854,61822005).
### Run 0 ended (1208.71s).
Writing output to remollout.root ... done
Graphics systems deleted.
Visualization Manager deleting...
 Running time[s]: 1231.85


**For 'feature-collimator2-update' branch** 
G4WT1 > --> Event 9000 starts with initial seeds (45799854,61822005).
### Run 0 ended (1208.59s).
Writing output to remollout.root ... done
Graphics systems deleted.
Visualization Manager deleting...
 Running time[s]: 1233.91


I don't see any significant degradation of running time...
I am open to discuss about implementing it better way..
But I have many other pending jobs.. If someone agreed volunteer, then I can give the details of the geometry.

Comment (wdconinc) [reply geometry/upstream/upstreamDaughter_merged.gdml:532]: I don't know if it was ok previously. When I last checked (cubic interpolation) we spent most of the simulation time dealing with polycones. Not sure if those are collimators.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**